pkg/manifest/os.go

```diff
@@ -768,6 +774,9 @@ func (p *OS) serialize() osbuild.Pipeline {
 				"wheel", "docker",
 			},
 		}))
+		if p.Bootupd {
```

Comment:
I would love to have a test for this just to make sure that if this option is specified we do not regress in generating the stage. I know it sounds a bit silly but I can see a future in which we refactor this function a bit and then having tests that check the high level behavior really works are useful.

Reply:
So, in theory, the manifests that we generate as part of CI should be the way to check that changes / refactoring don't have unintended effect. It is easy to do locally as part of development. It is not so easy to see in the CI itself. In the past, we used to have `diff-manifests` job in the osbuild-composer CI, but the job was probably too noisy and mostly ignored... :confused:

Reply:
Yeah, there's no good tradeoff here.  We can't have a test that fails when manifests change because that happens a lot.  We can't have people manually approve manifest changes because that gets too noisy and we just end up approving blindly.
The test we do have now uses bootc-image-builder to make sure that the container image can be built into a bootable disk image and that test will run if anything changes in the image definition (and soon also whenever BIB changes).  This kind of functional test is great IMO, but it doesn't help with catching things during development.  For that, generating manifests locally and checking a diff is the most convenient way.

Reply:
I think at this level it is sufficient to check that the stage is included when the option is set. This is what the `TestBootupdStage` in `[0001-osbuild-manifest-add-tests-for-bootupd-stage.patch.txt](https://github.com/osbuild/images/files/14193724/0001-osbuild-manifest-add-tests-for-bootupd-stage.patch.txt)` is doing. Or am I missing something why we might not want this?

Reply:
Sorry, I'm surely stating the obvious (again) but the way I see it is that the tests are build on top of each others. So here we ensure that the stage is included, elsewhere we ensure the stage is generated and elsewhere that a stage is transformed into correct json. So here we test just that, in the other test `TestNewBootupdGenMetadataStage` we ensure that the right stage is generated and  in stages_test.go we have a test that ensures that `Type` for stages ends up in the right shape for consumption by osbuild. With that we have all we need here and all tests are fast :)